src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Embeds the App Name into the AppHost.exe  
+    /// </summary>
+    public class EmbedAppNameInHost : TaskBase
+    {
+        private ITaskItem _outputAppHostPath;
+
+        [Required]
+        public string AppHostSourcePath { get; set; }
+
+        [Required]
+        public string AppHostDestinationDirectoryPath { get; set; }
+
+        [Required]
+        public string AppName { get; set; }
+
+        [Output]
+        public ITaskItem ModifiedAppHostPath
+        {
+            get { return _outputAppHostPath; }
+        }
+
+        protected override void ExecuteCore()
+        {
+            var array = File.ReadAllBytes(AppHostSourcePath);
+            var hostExtension = Path.GetExtension(AppHostSourcePath);
+           
+            var appbaseName = Path.GetFileNameWithoutExtension(AppName);
+            var destinationDirectory = Path.GetFullPath(AppHostDestinationDirectoryPath);
+            var modifiedAppHostPath = Path.Combine(destinationDirectory, $"{appbaseName}{hostExtension}");
+            _outputAppHostPath= new TaskItem(modifiedAppHostPath);
+
+            if ( File.Exists(modifiedAppHostPath))
+            {
+                //We have already done the required modification to apphost.exe
+                return;
+            }
+
+            if (AppName.Length > 1024)
+            {
+                throw new BuildErrorException(Strings.FileNameIsTooLong, AppName);
+            }
+
+            var placeHolder = "c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2"; //hash value embedded in apphost.exe
```

Comment:
It will be good to add more comments around this and pointers to areas in Core-Setup to which this is connect. It will help the reader with more context.